packages/shared/src/hooks/useChangelog.ts

```diff
@@ -1,27 +1,65 @@
 import { useContext, useMemo } from 'react';
-import { useMutation, useQuery } from 'react-query';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
 import AlertContext from '../contexts/AlertContext';
 import { getLatestChangelogPost, Post } from '../graphql/posts';
 import useSidebarRendered from './useSidebarRendered';
+import { ToggleVoteProps, useVotePost, voteMutationHandlers } from './vote';
 
 type UseChangelog = {
   isAvailable: boolean;
   latestPost?: Post;
   dismiss: () => Promise<void>;
+  toggleUpvote: (props: ToggleVoteProps) => Promise<void>;
 };
 
 export function useChangelog(): UseChangelog {
   const { alerts, updateAlerts } = useContext(AlertContext);
   const { sidebarRendered } = useSidebarRendered();
+  const client = useQueryClient();
+
+  const changelogQueryKey = ['changelog', 'latest-post'];
```

Comment:
For query keys we should always aim to use `generateQueryKey` function it will ensure us the order is always matching expectations.

Reply:
Also, you can add `RequestKey.Changelog` to be used instead of `changelog` string.

Reply:
Where would 'latest-post' fit there? Something like this:
`generateQueryKey(RequestKey.Changelog, null, RequestKey.LatestPost)`

Reply:
`latestPost` can be just a string, since it's a sub resource if that makes sense. We use `RequestKey` only for top level resource so we can more easily invalidate.

Reply:
Other then that yeah, exactly like you did it 👌

Reply:
Cool, thank you!!